Scripts/UI/Base/UserInterface.gd

```diff
@@ -0,0 +1,103 @@
+extends Control
+class_name UserInterface
+## UserInterface Class: [br]
+## A Base UI class to contain common UI features within the game such as:[br]
+## - Controller Ui Support
+## - viewport Text Reizing (currently requires text size override to be on.).[br]
+## - Translation Switching (Unimplemented).
+
+@export_group("Focus Start", "start")
+
+enum FOCUS_START_ENUM 
+{
+	## Activates when ready function is called.
+	ready,
+	## Activates when changed to visible.
+	visibility_changed,
+	## Activates when either ready or visible is called
+	both
+}
+
+@export var start_on: bool = true
+@export var start_focus: Control
+@export var start_MODE: FOCUS_START_ENUM
+@export_group("Text Resize", "resize")
+
+## Turns Text Resizing on or off.
```

Comment:
Documentation for vars should be inline with the variable declaration.

Reply:
fixed